tests/templates.test.js

```diff
@@ -0,0 +1,118 @@
+import { createMocks } from "node-mocks-http";
+import templates from "../pages/api/templates";
+import client from "next-auth/client";
+import validFormTemplate from "./validFormTemplate.json";
+import brokenFormTemplate from "./brokenFormTemplate.json";
+
+import fetchMock from "jest-fetch-mock";
+
+global.TextEncoder = require("util").TextEncoder;
+global.TextDecoder = require("util").TextDecoder;
+
+jest.mock("next-auth/client");
+
+jest.mock("@aws-sdk/client-lambda", () => {
+  return {
+    LambdaClient: jest.fn(() => {
+      return {
+        send: jest.fn((command) => {
+          const encoder = new TextEncoder(),
+            decoder = new TextDecoder();
+          const payload = JSON.parse(decoder.decode(command.Payload));
+          const supported = ["GET", "UPDATE", "INSERT", "DELETE"];
+
+          if (supported.indexOf(payload.method) === -1) {
+            return Promise.resolve({
+              FunctionError: "Unsupported",
+            });
+          } else if (payload.formID === "1") {
+            return Promise.resolve({
+              Payload: encoder.encode(
+                JSON.stringify({
+                  data: {
+                    records: [
+                      {
+                        formConfig: {
+                          submission: {
+                            vault: true,
+                          },
+                        },
+                      },
+                    ],
+                  },
+                })
+              ),
+            });
+          } else {
+            return Promise.resolve({
+              Payload: encoder.encode(
+                JSON.stringify({
+                  data: { records: [{ formID: 8 }] },
+                })
+              ),
+            });
+          }
+        }),
+      };
+    }),
+    InvokeCommand: jest.fn((payload) => {
+      return payload;
+    }),
+  };
+});
+
+describe("Test JSON validation scenarios", () => {
```

Comment:
Overall these tests look good ! Well done ! There are some additional test cases like for example the lambda client throwing an error but this is outside the scope of this PR